### PR TITLE
[function](json) Json unquote

### LINKED ISSUE #17216: [Feature] add some new json function
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

`json_contains` : [https://dev.mysql.com/doc/refman/5.7/en/json-search-functions.html#function_json-contains](https://dev.mysql.com/doc/refman/5.7/en/json-search-functions.html#function_json-contains)

`json_extract` : support return array, like : [https://dev.mysql.com/doc/refman/5.7/en/json-search-functions.html#function_json-extract](https://dev.mysql.com/doc/refman/5.7/en/json-search-functions.html#function_json-extract)

`json_unqoute` : [https://dev.mysql.com/doc/refman/5.7/en/json-modification-functions.html#function_json-unquote](https://dev.mysql.com/doc/refman/5.7/en/json-modification-functions.html#function_json-unquote)

`json_valid` : [https://dev.mysql.com/doc/refman/5.7/en/json-attribute-functions.html#function_json-valid](https://dev.mysql.com/doc/refman/5.7/en/json-attribute-functions.html#function_json-valid)

`-> operator` : [https://dev.mysql.com/doc/refman/5.7/en/json-search-functions.html#operator_json-column-path](https://dev.mysql.com/doc/refman/5.7/en/json-search-functions.html#operator_json-column-path)

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Mellorsssss**
# Proposed changes

Issue Number: close #17216

## Problem summary

Implement json_unquote.
```
mysql> SELECT json_unquote('"doris"');
+-------------------------+
| json_unquote('"doris"') |
+-------------------------+
| doris                   |
+-------------------------+
```

## Checklist(Required)

* [ ] Does it affect the original behavior
* [x] Has unit tests been added
* [x] Has document been added or modified
* [ ] Does it need to update dependencies
* [x] Is this PR support rollback (If NO, please explain WHY)

## Further comments

Currently, the [NO_BACKSLASH_ESCAPES](https://dev.mysql.com/doc/refman/5.7/en/sql-mode.html#sqlmode_no_backslash_escapes) sql mode specified in the MySQL documentation at  does not actually take effect. Therefore, even if a user enables NO_BACKSLASH_ESCAPES, the behavior of json_unquote will not change.

### CODE REVIEW

**Review by Yukang-Lian — DISMISSED** [review]
LGTM

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by dataroaring** [comment]
run buildall

**Comment by dataroaring** [comment]
add a regression test for normal string and "".

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 33.13 seconds
 stream load tsv:          447 seconds loaded 74807831229 Bytes, about 159 MB/s
 stream load json:         24 seconds loaded 2358488459 Bytes, about 93 MB/s
 stream load orc:          73 seconds loaded 1101869774 Bytes, about 14 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230325151441_clickbench_pr_120626.html

File: be/src/vec/functions/function_json.cpp [hunk 1]
```diff
@@ -754,11 +754,80 @@ class FunctionJsonValid : public IFunction {
...
+        rapidjson::Document document;
+        for (size_t i = 0; i < input_rows_count; ++i) {
+            if (col_from.is_null_at(i)) {
+                null_map->get_data()[i] = 1;
+                col_to->insert_data(nullptr, 0);
+                continue;
+            }
+
+            const auto& data = col_from_string->get_data_at(i);
+            if (data.size < 2 || data.data[0] != '"' || data.data[data.size - 1] != '"') {
```
Comment (xiaokang) [inline be/src/vec/functions/function_json.cpp:804]: If data.size == 0, data[0] should not be used.

File: be/src/vec/functions/function_json.cpp [hunk 2]
```diff
@@ -754,11 +754,80 @@ class FunctionJsonValid : public IFunction {
...
+        // parser can be reused for performance
+        rapidjson::Document document;
+        for (size_t i = 0; i < input_rows_count; ++i) {
+            if (col_from.is_null_at(i)) {
+                null_map->get_data()[i] = 1;
+                col_to->insert_data(nullptr, 0);
+                continue;
+            }
+
+            const auto& data = col_from_string->get_data_at(i);
```
Comment (xiaokang) [inline be/src/vec/functions/function_json.cpp:803]: The name data.data is strange. Can you change variable data to json_str?

**Comment by Mellorsssss** [comment]
done

**Comment by Mellorsssss** [comment]
run buildall

File: be/src/vec/functions/function_json.cpp (hunk 1 above)
Comment (Mellorsssss) [reply be/src/vec/functions/function_json.cpp:804]: If the data.size == 0, then data.data[0] will not be evaluated since data.size < 2 will be true. I add a test case for the empty string ''.

File: be/src/vec/functions/function_json.cpp (hunk 2 above)
Comment (Mellorsssss) [reply be/src/vec/functions/function_json.cpp:803]: Thanks xiaokang! I will rename it.

**Comment by Mellorsssss** [comment]
run buildall

**Comment by Mellorsssss** [comment]
run buildall

**Comment by Mellorsssss** [comment]
run buildall

**Comment by nextdreamblue** [comment]
LGTM

**Review by xiaokang — APPROVED** [review]
LGTM

**Comment by Yukang-Lian** [comment]
run p1

**Review by dataroaring — APPROVED** [review]
LGTM
